redfish-core/lib/cable.hpp

```diff
@@ -60,17 +99,236 @@ inline void
             // Cable length is NaN by default, do not throw an error
             if (!std::isnan(*length))
             {
-                messages::internalError(resp);
+                messages::internalError(asyncResp->res);
                 return;
             }
         }
         else
         {
-            resp.jsonValue["LengthMeters"] = *length;
+            asyncResp->res.jsonValue["LengthMeters"] = *length;
         }
     }
 }
 
+/**
+ * @brief Fill cable specific properties.
+ * @param[in,out]   asyncResp       Async HTTP response.
+ * @param[in]       ec          Error code corresponding to Async method call.
+ * @param[in]       properties  List of Cable Properties key/value pairs.
+ */
+inline void
+    fillCableProperties(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& cableObjectPath,
+                        const std::string& service)
+{
+    sdbusplus::asio::getAllProperties(
+        *crow::connections::systemBus, service, cableObjectPath,
+        "xyz.openbmc_project.Inventory.Item.Cable",
+        [asyncResp](const boost::system::error_code& ec,
+                    const dbus::utility::DBusPropertiesMap& properties) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error {}", ec.value());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        afterFillCableProperties(asyncResp, properties);
+    });
+}
+
+inline void
+    fillCableHealthState(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const std::string& cableObjectPath,
+                         const std::string& service)
+{
+    sdbusplus::asio::getProperty<bool>(
+        *crow::connections::systemBus, service, cableObjectPath,
+        "xyz.openbmc_project.Inventory.Item", "Present",
+        [asyncResp, cableObjectPath](const boost::system::error_code& ec,
+                                     bool present) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("get presence failed for Cable {} with error {}",
+                             cableObjectPath, ec.value());
+            if (ec.value() != EBADR)
+            {
+                messages::internalError(asyncResp->res);
+            }
+            return;
+        }
+
+        if (!present)
+        {
+            asyncResp->res.jsonValue["Status"]["State"] = "Absent";
+        }
+    });
+}
+
+inline void getCableUpstreamResources(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& cableObjectPath)
+{
+    // retrieve Upstream Resources
+    std::string upstreamResource = cableObjectPath + "/upstream_resource";
+    linkAssociatedCable(
+        asyncResp, upstreamResource,
+        [asyncResp](const dbus::utility::MapperEndPoints& endpoints) {
```

Comment:
Should this > 10 line lambda be in a named function?

Reply:
I'm still wondering how this part of logic can be simplified...

Reply:
I refactored those lambda usage cases.